testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy

```diff
@@ -306,6 +308,11 @@ class BuildScanPluginSmokeTest extends AbstractSmokeTest {
 
         expect:
         scanRunner("--init-script", initScript)
+            .maybeExpectLegacyDeprecationWarningIf(FIRST_VERSION_UNDER_DEVELOCITY_BRAND <= versionNumber,
+                "WARNING: The following functionality has been deprecated and will be removed in the next major release of the Develocity Gradle plugin:")
+            .maybeExpectLegacyDeprecationWarningIf(FIRST_VERSION_UNDER_DEVELOCITY_BRAND <= versionNumber,
+                "WARNING: The following functionality has been deprecated and will be removed in the next major release of the Develocity Gradle plugin. " +
+                    "For assistance with migration, see https://gradle.com/help/gradle-plugin-develocity-migration.")
```

Comment:
I suppose we should update the git refs for the init scripts we test in a follow-up PR: https://github.com/gradle/gradle/blob/1a5e48701b602def428d9c79e3793adff7a4f798/testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy#L193-L201

Reply:
You are right. I would have done it in this PR, but the CI plugin versions supporting Develocity have not been released yet.